core/src/vaults.ts

```diff
@@ -0,0 +1,446 @@
+});
+
+const _fetchLiquidatedParameters = async (network: string, vault: Vault) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const typeHex = ethers.utils.formatBytes32String(vault.collateralType);
+    const eventFilter = contract.filters.Bark(typeHex, vault.address, null, null, null, null, null);
+    const liquidationEvents = await contract.queryFilter(eventFilter);
+    if (liquidationEvents.length !== 0 && vault.initialDebtDai.eq(0)) {
+        // there was a liquidation and the vault was not used again
+        return Promise.all(
+            liquidationEvents.map(async event => ({
+                liquidationDate: await fetchDateByBlockNumber(network, event.blockNumber),
+                transactionHash: event.transactionHash,
+                auctionId: `${vault.collateralType}:${new BigNumber(event.args?.id._hex).toFixed(0)}`,
+            }))
+        );
+    }
+    return undefined;
+};
+export const fetchLiquidatedParameters = memoizee(_fetchLiquidatedParameters, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const getAuctionedDaiAndAuctionState = (proximityToLiquidation: BigNumber, vault: Vault, debtDai: BigNumber) => {
+    // logic from https://github.com/makerdao/dss/blob/fa4f6630afb0624d04a003e920b0d71a00331d98/src/dog.sol#L186
+    // detemines if the vault is liquidatable and what amount of debt can be covered.
+    let state: 'liquidatable' | 'not-liquidatable' = proximityToLiquidation.isLessThanOrEqualTo(0)
+        ? 'liquidatable'
+        : 'not-liquidatable';
+    const amountDaiCanBeAuctionedGloballyDai = vault.maximumProtocolDebtDai.minus(vault.currentProtocolDebtDai);
+    const amountDaiCanBeAuctionedCollateralDai = vault.maximumCollateralDebtDai.minus(vault.currentCollateralDebtDai);
+    const minimumDebtCovered = amountDaiCanBeAuctionedCollateralDai.isLessThan(amountDaiCanBeAuctionedGloballyDai)
+        ? amountDaiCanBeAuctionedCollateralDai
+        : amountDaiCanBeAuctionedGloballyDai;
+    if (amountDaiCanBeAuctionedGloballyDai.eq(0) || amountDaiCanBeAuctionedCollateralDai.eq(0)) {
+        state = 'not-liquidatable';
+    }
+    let auctionedAmountDai = BigNumber.min(
+        vault.initialDebtDai,
+        minimumDebtCovered.div(vault.stabilityFeeRate).div(vault.liquidationPenaltyRatio)
+    );
+    if (auctionedAmountDai.isLessThan(vault.initialDebtDai)) {
+        if (
+            vault.initialDebtDai
+                .minus(auctionedAmountDai)
+                .multipliedBy(vault.stabilityFeeRate)
+                .isLessThan(vault.minimalAuctionedDai)
+        ) {
+            auctionedAmountDai = debtDai;
+        } else {
+            if (
+                auctionedAmountDai
+                    .multipliedBy(vault.stabilityFeeRate)
+                    .isGreaterThanOrEqualTo(vault.minimalAuctionedDai)
+            ) {
+                state = 'not-liquidatable';
+            }
+        }
+    }
+    return { state, auctionedAmountDai };
```

Comment:
What is `auctionedAmountDai`? Amount of DAI that will be auctioned in case of the partial auction which can be smaller than `debtDai`? Maybe `debtThatCanBeAuctionedDai`?

I suggest to split this into two functions for readability: one that computes `debtThatCanBeAuctionedDai`, the other one which determines state checking if `debtThatCanBeAuctionedDai` bigger than 0 and the minimum

Reply:
Also, I don't know if you know noticed it, but limits also account for the incentives

Reply:
> debtThatCanBeAuctionedDai

actually `initialDebtThatCanBeAuctionedDai`

Reply:
> Also, I don't know if you know noticed it, but limits also account for the incentives

this seems out of context and i can't figure what you're referring to here. could you elaborate?

Reply:
> limits also account for the incentives

Do you account for combined incentive when determining if auction will be executed partially? Because when vault is liquidated the limits are increased not only on the amount of debt, but also on the amount of incentives. Do you account for it or am I wrong?